### PR TITLE
chore: remove unused feature "document ranks"

### PR DESCRIPTION
**Pull request by stefanhengl**
Document ranks was an experimental feature that allowed Zoekt to query Sourcegraph for additional ranking signals for files at index-time. We have decided to remove this feature because it was never used in production.

Note:
Most of the changes are Sourcegraph specific. However there are some exported fields in `SearchOptions` and `build.Options` as well as flags for zoekt-git-index which I simply removed. I believe it is fairly unlikely that those were used by anyone outside Sourcegraph.

Test plan:
updated tests

### CODE REVIEW

File: cmd/zoekt-sourcegraph-indexserver/protos/sourcegraph/zoekt/configuration/v1/configuration.proto [hunk 1]
```diff
@@ -13,9 +13,6 @@ service ZoektConfigurationService {
   // List returns the list of repositories that the client should index.
   rpc List(ListRequest) returns (ListResponse) {}
 
-  // DocumentRanks returns the rank vectors for all documents in the specified repository.
-  rpc DocumentRanks(DocumentRanksRequest) returns (DocumentRanksResponse) {}
```
Comment (stefanhengl) [inline cmd/zoekt-sourcegraph-indexserver/protos/sourcegraph/zoekt/configuration/v1/configuration.proto:17]: This has not been called for a long time, so it should be save to remove.

File: cmd/zoekt-git-index/main.go [hunk 2]
```diff
@@ -43,8 +43,6 @@ func run() int {
 		"It also affects name if the indexed repository is under this directory.")
 	isDelta := flag.Bool("delta", false, "whether we should use delta build")
 	deltaShardNumberFallbackThreshold := flag.Uint64("delta_threshold", 0, "upper limit on the number of preexisting shards that can exist before attempting a delta build (0 to disable fallback behavior)")
-	offlineRanking := flag.String("offline_ranking", "", "the name of the file that contains the ranking info.")
-	offlineRankingVersion := flag.String("offline_ranking_version", "", "a version string identifying the contents in offline_ranking.")
```
Comment (stefanhengl) [inline cmd/zoekt-git-index/main.go:47]: This is backward incompatible but I believe it's very unlikely that anyone used this outside Sourcegraph.

**Review by jtibshirani — APPROVED** [review]
Nice clean-up. To check I understand, this will *not* trigger a reindex for everyone right? Because customers had an empty `DocumentRanksVersion` all along (we never enabled the feature for them).

File: eval.go [hunk 3]
```diff
@@ -386,11 +386,6 @@ nextFileMatch:
 	// Update stats based on work done during document search.
 	updateMatchTreeStats(mt, &res.Stats)
 
-	// If document ranking is enabled, then we can rank and truncate the files to save memory.
-	if opts.UseDocumentRanks {
```
Comment (jtibshirani) [inline eval.go:390]: This seems weird to me. Can't we always sort and truncate the files here?

**Comment by stefanhengl** [comment]
closing this in favor of #853

**Pull request closed without merging**